### DISCUSSION TITLE
{{linksection}} and/or {{linkblock}} in addition to {{linkcurrent}}?

### DISCUSSION THREAD

**Discussion by tauren** [chhoumann/quickadd#380] [Ideas]
With `{{linkcurrent}}`, a link is created to the current file, but clicking that link opens the file scrolled to the top. This makes it difficult to maintain context, especially in larger files.

It would be great to create a link to the current section or block that the cursor is in. Perhaps adding a `{{linksection}}` or `{{linkblock}}` feature might make sense?

**Comment by chhoumann** [comment]
Could be a pretty good addition. I've made a feature request for this. #387

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

`{{LINKSECTION}}` shipped in v2.14.0 (#387). It works like `{{LINKCURRENT}}`, but links to the heading your cursor is under, in `[[Note#Heading]]` form - so clicking the link scrolls to the section instead of the top of the file.

```markdown
Source: {{LINKSECTION}}
```

To be upfront: `{{linkblock}}` (block-reference granularity) didn't get built. If you'd still find that useful, a focused feature request is welcome.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#linksection

Closing as resolved.
